mnist/mnist_gcp.ipynb

```diff
@@ -0,0 +1,895 @@
+    "    matchLabels:\n",
+    "      app: mnist-tensorboard\n",
+    "  template:\n",
+    "    metadata:\n",
+    "      labels:\n",
+    "        app: mnist-tensorboard\n",
+    "        version: v1\n",
+    "    spec:\n",
+    "      serviceAccount: default-editor\n",
+    "      containers:\n",
+    "      - command:\n",
+    "        - /usr/local/bin/tensorboard\n",
+    "        - --logdir={model_dir}\n",
+    "        - --port=80\n",
+    "        image: tensorflow/tensorflow:1.15.2-py3\n",
+    "        name: tensorboard\n",
+    "        ports:\n",
+    "        - containerPort: 80\n",
+    "\"\"\"\n",
+    "tb_service = f\"\"\"apiVersion: v1\n",
+    "kind: Service\n",
+    "metadata:\n",
+    "  labels:\n",
+    "    app: mnist-tensorboard\n",
+    "  name: {tb_name}\n",
+    "  namespace: {namespace}\n",
+    "spec:\n",
+    "  ports:\n",
+    "  - name: http-tb\n",
+    "    port: 80\n",
+    "    targetPort: 80\n",
+    "  selector:\n",
+    "    app: mnist-tensorboard\n",
+    "  type: ClusterIP\n",
+    "\"\"\"\n",
+    "\n",
+    "tb_virtual_service = f\"\"\"apiVersion: networking.istio.io/v1alpha3\n",
+    "kind: VirtualService\n",
+    "metadata:\n",
+    "  name: {tb_name}\n",
+    "  namespace: {namespace}\n",
+    "spec:\n",
+    "  gateways:\n",
+    "  - kubeflow/kubeflow-gateway\n",
+    "  hosts:\n",
+    "  - '*'\n",
+    "  http:\n",
+    "  - match:\n",
+    "    - uri:\n",
+    "        prefix: /mnist/{namespace}/tensorboard/\n",
+    "    rewrite:\n",
+    "      uri: /\n",
+    "    route:\n",
+    "    - destination:\n",
+    "        host: {tb_name}.{namespace}.svc.cluster.local\n",
+    "        port:\n",
+    "          number: 80\n",
+    "    timeout: 300s\n",
+    "\"\"\"\n",
+    "\n",
+    "tb_specs = [tb_deploy, tb_service, tb_virtual_service]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "k8s_util.apply_k8s_specs(tb_specs, k8s_util.K8S_CREATE_OR_REPLACE)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Access The TensorBoard UI"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "endpoint = k8s_util.get_iap_endpoint() \n",
+    "if endpoint:    \n",
+    "    vs = yaml.safe_load(tb_virtual_service)\n",
+    "    path= vs[\"spec\"][\"http\"][0][\"match\"][0][\"uri\"][\"prefix\"]\n",
+    "    tb_endpoint = endpoint + path\n",
+    "    display(HTML(f\"TensorBoard UI is at <a href='{tb_endpoint}'>{tb_endpoint}</a>\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wait For the Training Job to finish"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tf_job = tf_job_client.wait_for_condition(train_name, expected_condition=[\"Succeeded\", \"Failed\"], namespace=namespace)"
```

Comment:
could this add a `print(tf_job)` or similar so that they can see if the job succeeded or failed?

Reply:
Done.